substrafl/model_loading.py

```diff
@@ -242,71 +337,138 @@ def download_algo_files(
     folder = Path(dest_folder)
     folder.mkdir(exist_ok=True, parents=True)
 
-    if round_idx is None:
-        round_idx = compute_plan.metadata["num_rounds"]
-
-    # Get the train task associated to user inputs
-    train_task = _get_train_task_from_round(client=client, compute_plan_key=compute_plan_key, round_idx=round_idx)
-
-    if train_task.status is not Status.done:
-        raise UnfinishedTrainTaskError(
-            f"Can't download algo files form task {train_task.key} as it is " f"in status {train_task.status}"
+    # Get the task associated to user inputs
+    if round_idx is not None and rank_idx is not None:
+        raise exceptions.ArgumentConflictError("Only one out of round_idx and rank_idx must be specified.")
+    elif round_idx is not None:
```

Comment:
Minor but I would restart with a regular `if` here, given that we have raised in the other case.